CMakeLists.txt

```diff
@@ -39,6 +39,11 @@ check_cxx_compiler_flag(-fopenmp-simd CXX_HAVE_OMP_SIMD)
 # Build options
 option(BUILD_DALI_NODEPS "Disable components that require extra external libraries to be present in the system. Effectively, it builds only the DALI core and kernel libraries")
 option(LINK_DRIVER "Links directly with libcuda.so instead of dlopen it at runtime" OFF)
+option(WITH_DYNAMIC_CUDA_TOOLKIT "Links CUDA toolkit libraires dynamically (NPP, nvJPEG, cuFFT)" OFF)
+
+if (WITH_DYNAMIC_CUDA_TOOLKIT)
+  add_definitions(-DWITH_DYNAMIC_CUDA_TOOLKIT=1)
```

Comment:
Can you consider making it similar to `propagate_option` which exports 1 and 0 valued define in both cases?